app/controllers/concerns/token_auth.rb

```diff
@@ -34,7 +34,7 @@ def validate_access
       sign_in(token.user)
     end
     # expire tokens regardless of how user logged in
-    tokens = ApiToken.joins(:approval).where(approvals: {
+    tokens = ApiToken.joins(:step).where(steps: {
       user_id: current_user, proposal_id: self.proposal})
```

Comment:
I *think* you could just do `where( steps: { user: current_user, proposal: proposal })`